### PR TITLE
chore(linter): change tslint to eslint with typescript support

### PR DESCRIPTION
**Pull request by wermanoid**
moving to eslint-typescript and fix errors

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+dist/
```
Comment (serhii-havrylenko) [inline .eslintignore]: Newline plz

Comment (wermanoid) [reply .eslintignore]: updated

File: .eslintrc.prod.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
```
[the remark this answers is not in the document]
Comment (wermanoid) [reply .eslintrc.prod.js]: yeah. updated

File: .npmrc [hunk 3]
```diff
@@ -1 +1 @@
-; registry=http://localhost:4873
\ No newline at end of file
+registry=http://localhost:4873
```
[the remark this answers is not in the document]
Comment (wermanoid) [reply .npmrc]: in place

File: jest.config.js [hunk 4]
```diff
@@ -27,5 +27,5 @@ module.exports = {
   transform: {
     '^.+\\.(ts|tsx)$': 'ts-jest',
   },
-  setupFilesAfterEnv: ['<rootDir>jest/setupTests.ts'],
+  setupFilesAfterEnv: ['@testing-library/jest-dom/extend-expect'],
```
[the remark this answers is not in the document]
Comment (wermanoid) [reply jest.config.js]: restored

**Review by sebastian-lenart — APPROVED** [review]

**Pull request merged**